src/riak_kv_wm_timeseries.erl

```diff
@@ -29,572 +34,382 @@
-            {F, GoodValue}
-    end.
-
-ensure_lk_order_and_strip(LK, FVList) ->
-    [proplists:get_value(F, FVList)
-     || #param_v1{name = F} <- LK].
-
-
--spec valid_params(#wm_reqdata{}, #ctx{}) -> ?CB_RV_SPEC.
-valid_params(RD, Ctx) ->
-    case wrq:get_qs_value("timeout", none, RD) of
-        none ->
-            {true, RD, Ctx};
-        TimeoutStr ->
-            try
-                Timeout = list_to_integer(TimeoutStr),
-                {true, RD, Ctx#ctx{timeout = Timeout}}
-            catch
-                _:_ ->
-                    handle_error({bad_parameter, "timeout"}, RD, Ctx)
-            end
-    end.
-
-
--spec check_permissions(#wm_reqdata{}, #ctx{}) -> {term(), #wm_reqdata{}, #ctx{}}.
-check_permissions(RD, Ctx = #ctx{security = undefined}) ->
-    {true, RD, Ctx};
-check_permissions(RD, Ctx = #ctx{security = Security,
-                                 api_call = Call,
-                                 table = Table}) ->
-    case riak_core_security:check_permission(
-           {riak_kv_ts_util:api_call_to_perm(Call), Table}, Security) of
-        {false, Error, _} ->
-            handle_error(
-              {not_permitted, utf8_to_binary(Error)}, RD, Ctx);
+        GoodValue when GoodValue > 0 ->
+            GoodValue;
         _ ->
-            {true, RD, Ctx}
+            throw({path_error, "incorrect field value"})
     end.
 
-
--spec content_types_provided(#wm_reqdata{}, #ctx{}) ->
-                                    {[{ContentType::string(), Producer::atom()}],
-                                     #wm_reqdata{}, #ctx{}}.
-content_types_provided(RD, Ctx) ->
-    {[{"application/json", produce_doc_body}], RD, Ctx}.
-
-
--spec encodings_provided(#wm_reqdata{}, #ctx{}) ->
-                                {[{Encoding::string(), Producer::function()}],
-                                 #wm_reqdata{}, #ctx{}}.
-encodings_provided(RD, Ctx) ->
-    {riak_kv_wm_utils:default_encodings(), RD, Ctx}.
-
-
--spec content_types_accepted(#wm_reqdata{}, #ctx{}) ->
-                                    {[{ContentType::string(), Acceptor::atom()}],
-                                     #wm_reqdata{}, #ctx{}}.
-content_types_accepted(RD, Ctx) ->
-    {[{"application/json", accept_doc_body}], RD, Ctx}.
-
-
--spec resource_exists(#wm_reqdata{}, #ctx{}) ->
-                             {boolean(), #wm_reqdata{}, #ctx{}}.
-resource_exists(RD, Ctx = #ctx{table = Table}) ->
-    Mod = riak_ql_ddl:make_module_name(Table),
+extract_data(RD, Mod) ->
     try
-        DDL = Mod:get_ddl(),
-        {true, RD, Ctx#ctx{mod = Mod, ddl = DDL}}
+        JsonStr = binary_to_list(wrq:req_body(RD)),
+        Json = mochijson2:decode(JsonStr),
+        DDLFieldTypes = ddl_fields_and_types(Mod),
+        extract_records(Json, DDLFieldTypes)
     catch
-        error:undef ->
-            handle_error({no_such_table, Table}, RD, Ctx)
+        Error:Reason ->
+            lager:log(info, self(), "extract_data: ~p:~p", [Error, Reason]),
```

Comment:
I don't think we should log here or a problem on a production system would fill the logs.  Error should be returned to the client, and could be included in an audit log.